nio_cli/utils.py

```diff
@@ -0,0 +1,134 @@
+                tmp.write('WS_HOST={}\n'.format(ws_host))
+            elif re.search('^PK_TOKEN=', line) and pk_token:
+                tmp.write('PK_TOKEN={}\n'.format(pk_token))
+            else:
+                tmp.write(line)
+    os.remove(conf_location)
+    os.rename(tmp.name, conf_location)
+
+    secure = input('Optional secure instance configuration [y/N]: ')
+    if secure.lower() == 'y':
+        _config_ssl(name, conf_location)
+
+
+def _config_ssl(name, conf_location):
+
+    cwd = os.getcwd()
+
+    new_certs = input('Generate a self-signed certificate/key [y/N]: ')
+
+    if new_certs.lower() == 'y':
+        try:
+            from OpenSSL import crypto
+        except Exception as e:
+            print('No pyOpenSSL installation detected. Your instance has still '
+                  'been configured but no certs were installed. To install '
+                  'certificates install pyOpenSSL and re-run "nio config" from '
+                  'inside the project directory.')
+            return
+
+        # Create a key pair
+        kp = crypto.PKey()
+        kp.generate_key(crypto.TYPE_RSA, 2048)
+
+        # Create a self-signed cert
+        cert = crypto.X509()
+        cert.get_subject().C = input('Enter two-letter country code: ')
+        cert.get_subject().ST = input('Enter state: ')
+        cert.get_subject().L = input('Enter city: ')
+        cert.get_subject().O = input('Enter company/owner: ')
+        cert.get_subject().OU = input('Enter user: ')
+        cert.get_subject().CN = 'localhost'
+        cert.set_serial_number(1000)
+        cert.gmtime_adj_notBefore(0)
+        cert.gmtime_adj_notAfter(365 * 24 * 60 * 60)
+        cert.set_issuer(cert.get_subject())
+        cert.set_pubkey(kp)
+        cert.sign(kp, 'sha1')
+
+        open('{}/certificate.pem'.format(name), "wt").write(
+            str(crypto.dump_certificate(crypto.FILETYPE_PEM, cert)))
+        open('{}/private_key.pem'.format(name), "wt").write(
+            str(crypto.dump_privatekey(crypto.FILETYPE_PEM, kp)))
+
+        ssl_cert = '{}/{}/certificate.pem'.format(cwd, name)
+        ssl_key = '{}/{}/private_key.pem'.format(cwd, name)
+
+    else:
+        ssl_cert = input('Enter SSL certificate file location: ')
+        ssl_key = input('Enter SSL private key file location: ')
+
+    with open(conf_location, 'r') as nconf,\
+            tempfile.NamedTemporaryFile(mode='w', delete=False) as tmp:
+        for line in nconf:
+            if re.search('^ssl_certificate:', line) and ssl_cert:
+                tmp.write('ssl_certificate: {}\n'.format(ssl_cert))
+            elif re.search('^ssl_private_key:', line) and ssl_key:
+                tmp.write('ssl_private_key: {}\n'.format(ssl_key))
+            else:
+                tmp.write(line)
+    os.remove(conf_location)
+    os.rename(tmp.name, conf_location)
+
+
+def set_user(project_name, username, password):
+    # load users
+    users_location = '{}/etc/users.json'.format(project_name)
```

Comment:
This will need to write to `permissions.json` too right? I thought we said we were going to add it there with a default of `rwx` for this MVP

Reply:
Correct. I'll add that to the utils